addons/mail/tests/mail_tracking_duration_mixin_case.py

```diff
@@ -195,5 +195,5 @@ def _test_queries_batch_duration_tracking(self):
         self.flush_tracking()
         batch[self.track_duration_field] = self.stage_2.id
 
-        with self.assertQueryCount(2):
+        with self.assertQueryCount(4):
```

Comment:
If the issue did not break any test, maybe we could add a case (ideally, in one of the existing tests) to be sure it does not occur again in the future? :)

Reply:
Hello @Odoonan 

The test in which i have made changes was breaking because before my fix only 2 queries were running in that test but after my fix 4 queries will execute. So after the updation of this test case where i have added `4` in `self.assertQueryCount():` will test my fix.

Thanks!!

Reply:
Hello, I see. I meant that maybe we could add a small test case in one of the existing tests to ensure that the issue you fix do not break again later. (So a small test case that would break without your fix but not with it)

Cheers!

Reply:
Hello @Odoonan 
If i remove my fix i will get 2 queries executed in above test case which will not fail because in `assertQueryCount()` if the count is less than expected (i.e. 4 in our case) a logger info is generated from here - 
https://github.com/odoo/odoo/blob/ee31c84c360425b0acba4f04462bc752c121b94b/odoo/tests/common.py#L541-L544

So to add a case that would break without my fix but not with it i have to create a method `assertQueryCountEqual()` similar to `assertQueryCount()` in which the count should be equal to expected otherwise it will fail. So can i add a method like it in `odoo/community/odoo/tests/common.py`, Kindly suggest me!!

Reply:
Hello @sauc-odoo . I think there is a misunderstanding here, let me rephrase it. The `self.assertQueryCount(4)` is fine. You can keep it as indeed, there are more queries.

What I meant was to **also** add a unit test for the fix itself. For instance

1. create a record in a stage and 'wait' 5 minutes
2. Assert stage duration is set to 5 min
3. Change stage
4. Make sure the new duration is not 5 min

Something like that, do you see what I mean ? But actually it looks like such tests exist, so it is strange that they do not fail.

Eventually, do not update the code yet, I would like to have a deeper look first. :)

Cheers!